pontoon/sync/vcs/repositories.py

```diff
@@ -40,8 +42,16 @@ def pull(self, source=None, target=None):
         code, output, error = execute(command, target)
 
         if code == 0:
+
             log.debug("Git: Repository at " + source + " updated.")
 
+            if branch:
+                command = ["git", "checkout", branch]
+                code, output, error = execute(command)
```

Comment:
This command should be ran inside the checkout path, so add `target` as the second argument to `execute()`.